### ISSUE TITLE
Key focus not granted on OSX

### ISSUE THREAD

**Issue by seanhall1** [glfw/glfw#1515]
OS and version: OSX High Sierra v10.13.6
Release or commit: 3.3/b0796109629931b6fa6e449c15a177845256a407
Error messages: None
events output: None

I've been seeing an issue with my glfw project on OSX where occasionally when launching the program the window was shown, but I wasn't being given key focus. The window would show in front of all other windows on the screen, but the top bar was greyed out for the glfw window and for all other windows that I have open. It seems that no open window has key focus in this state. I was able to recreate this issue with the boing glfw example with a few lines of code change. I added a call to the pthread sleep function after the call to init() in the main function. I called sleep with a value of 20 seconds. Without the sleep command the boing example does not see this issue. I added the sleep call because my program does some preprocessing so there is some delay between creating the window and processing window events. It seems like if this delay goes takes longer than some amount of time, the glfw window does not receive a key event from OSX. To get out of this state I need to select a different window and then switch back to the glfw window. Then the glfw window will have key focus. It's likely this could be created with any glfw example. Boing was simply the first example I tried.

**Comment by orange451** [comment]
I've had this bug many times as well. Very frustrating as I have to alt-tab to gain focus

**Comment by bruvzg** [comment]
This type of bug can be caused by using bare executable without wrapping it into `.app` bundle. 

Bundling the app should help (it's possible to directly run executable within bundle and it should not have focus issues).

AFAIK the only way to "fix" focus issues in unbundled app is to simulate window focus switching (see https://github.com/godotengine/godot/pull/17187 for example).

**Comment by elmindreda** [comment]
I can reproduce this with unbundled applications but the boing example is bundled, so this may be a different issue.

**Comment by grant-h** [comment]
Yup still having this issue on raw GLFW applications. The trick is that you must be moving the mouse when launching to trigger this. It has a random chance of happening even then. I noticed because I gate the framerate based off of the `GLFW_FOCUSED` attribute. This was obvious as my app would lag even though I can see it in the foreground. The weirdest part is that I can still provide keyboard and mouse input to my application! ImGUI checks for the focused flag, so it will drop events, but normal GLFW callbacks trigger.

The indication that the application is in this state is the top dock showing the name of a different executable than the foregrounded one. This is most certainly an Apple bug that has been unfixed for quite a while.
